pkg/target/target.go

```diff
@@ -47,12 +48,31 @@ const wildcardNSPattern = `^(\*|\*-)?[a-z0-9]([-a-z0-9]*[a-z0-9])?$|^[a-z0-9]([-
 
 var _ handler.TargetHandler = &K8sValidationTarget{}
 
-type K8sValidationTarget struct{}
+type K8sValidationTarget struct {
+	cache *nsCache
```

Comment:
nit: You could make  `cache` a non-pointer, i.e. `cahce nsCache` to communicate that `K8sValidationTarget` is the "true" owner of the object. There's no performance benefit to having this field as a pointer, and we always want this field to exist.

(As with all nits, this is completely optional)